### DISCUSSION TITLE
where T : record

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#4754] [Q&A]
What is the right way to specify this method?
```
    internal static class WithExtensions {
        public static T WithBase<T>(this T This, ImportJobBase Base)
            where T : record, ICsvBase {

            var ret = This with {
                SourceId = Base.Source_Id,
                ResultId = Base.Result_Id,
                Created_At = Base.Dest_Created_At,
                Created_By = Base.Dest_Created_By,
                Updated_At = Base.Dest_Updated_At,
                Updated_By = Base.Dest_Updated_By,
            };

            return ret;
        }
    }
```

**Comment by HaloFour** [comment]
See: https://github.com/dotnet/csharplang/discussions/3859

**Comment by huoyaoyuan** [comment]
Currently there should be no way.
In future, specifying the `Clone` method and `init` accessors in the interface may work.

**Comment by jnm2** [comment]
If a library can tell the difference between classes I compiled using the `record` keyword and classes I compiled using the `class` keyword, that threatens my ability to be able to treat `record` vs `class` as an implementation detail that won't affect libraries.

I see the ask, though, which is less "can I tell if this is a record" and more "can I use withers on this thing." That's different and would not involve `where T : record`. The title of this discussion is going to skew the conversation in terms of my first point.
